### PR TITLE
Fix build for wasm32/64-unknown-unknown without std

### LINKED ISSUE #1527: Failed to build for wasm32-unknown-unknown with no_std
Hi,

I was playing with plonky2 for `wasm32-unknown-unknown` target without default features and found out that `plonky2_maybe_rayon` erroneously pulls `std` in.

Here is an example reproducing the issue.
<details><summary>Cargo.toml</summary>
<p>

```toml
[package]
name = "plonky2-mayberayon"
version = "0.1.0"
edition = "2021"

[dependencies]
sp-io = { version = "30.0.0", default-features = false }
plonky2_maybe_rayon = { git = "https://github.com/0xPolygonZero/plonky2.git", default-features = false }

[features]
default = ["std"]
std = ["sp-io/std"]
```

</p>
</details>

<details><summary>src/lib.rs</summary>
<p>

```rust
#![no_std]

extern crate sp_io;
extern crate plonky2_maybe_rayon;

```

</p>
</details>

```shell
cargo b --target=wasm32-unknown-unknown --no-default-features

...
   Compiling plonky2-mayberayon v0.1.0 (/home/user/devel/gear-tech/plonky2-mayberayon)
error[E0152]: duplicate lang item in crate `std` (which `plonky2_maybe_rayon` depends on): `panic_impl`.
  |
  = note: the lang item is first defined in crate `sp_io` (which `plonky2_mayberayon` depends on)
  = note: first definition in `sp_io` loaded from /home/user/devel/gear-tech/plonky2-mayberayon/target/wasm32-unknown-unknown/debug/deps/libsp_io-27b44a1938535aab.rmeta
  = note: second definition in `std` loaded from /home/user/.rustup/toolchains/nightly-2023-04-25-x86_64-unknown-linux-gnu/lib/rustlib/wasm32-unknown-unknown/lib/libstd-bdf48c82ab858a37.rlib

For more information about this error, try `rustc --explain E0152`.
error: could not compile `plonky2-mayberayon` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

### PR DESCRIPTION
**Pull request by gshep**
fixes #1527

### CODE REVIEW

File: maybe_rayon/src/lib.rs [hunk 1]
```diff
@@ -1,8 +1,6 @@
-#[cfg(not(feature = "parallel"))]
-use core::{
-    iter::{FlatMap, IntoIterator, Iterator},
-    slice::{Chunks, ChunksExact, ChunksExactMut, ChunksMut},
-};
+#![cfg_attr(not(std), no_std)]
+
+extern crate alloc;
```
Comment (Nashtare) [inline maybe_rayon/src/lib.rs:3]: We only need alloc conditionally imported here
```suggestion
#[cfg(not(feature = "parallel"))]
extern crate alloc;
```

File: plonky2/Cargo.toml [hunk 2]
```diff
@@ -31,7 +31,7 @@ plonky2_util = { path = "../util", default-features = false }
 rand = { version = "0.8.4", default-features = false }
 rand_chacha = { version = "0.3.1", optional = true, default-features = false }
 serde = { version = "1.0", default-features = false, features = ["derive", "rc"] }
-serde_json = "1.0"
+serde_json = { version = "1.0", default-features = false, features = ["alloc"] }
```
Comment (Nashtare) [inline plonky2/Cargo.toml:34]: This actually should be moved to `[dev-dependencies]`, it's not being used by the library (hence why it compiles fine against wasm targets), only by the `fibonacci_serialization` example which does require `std` anyway.

**Review by Nashtare — CHANGES_REQUESTED** [review]

**Review by Nashtare — APPROVED** [review]
LGTM! Thank you for fixing this!

**Comment by muraca** [comment]
I authored #1335 and I casually noticed this PR. I think `maybe_rayon` and all the other crates that should support `wasm32` should be added to the CI checks.

**Comment by Nashtare** [comment]
Yeah I've been telling myself this as well seeing this behavior. Will do shortly.

**Pull request merged**